File: frontend/src/api/content.js
import axios from 'axios';

const API_URL = 'http://localhost:8000/api';

export const generateContent = async (topic, subtopic, mode = "story", difficulty = "Normal", language = "English", images = null, videos = null, roadmapId = null, interest = null) => {
  try {
    const response = await axios.post(`${API_URL}/content/generate`, {
      topic,
      subtopic,
      mode,
      difficulty,
      language,
      images,
      videos,
      interest,
      roadmap_id: roadmapId
    });
    return response.data;
  } catch (error) {
    console.error("Error generating content:", error);
    throw error;
  }
};

File: backend/app/models/content.py
from pydantic import BaseModel
from typing import List, Optional, Any, Union

class ContentRequest(BaseModel):
    topic: str
    subtopic: str
    mode: str = "story" # story, deep, exam
    difficulty: str = "Normal"
    language: str = "English"
    interest: Optional[str] = None
    images: Optional[List[str]] = None
    videos: Optional[List[Union[str, dict]]] = None

class ContentResponse(BaseModel):
    content: str
    images: List[str] = []
    videos: List[Union[str, dict]] = []
    quiz_questions: List[dict] = []
